projects/js-packages/connection/components/connect-screen/required-plan/index.jsx

```diff
@@ -33,6 +34,8 @@ const ConnectScreenRequiredPlan = props => {
 		pricingIcon,
 		pricingTitle,
 		pricingCurrencyCode,
+		wpcomProductSlug,
+		checkSiteHasWpcomProduct,
```

Comment:
have you considered adding this function straight here?
I think the trickier part is the endpoint implementation.